### PR TITLE
[CDF-21433] 💬 Error message on wrong project.

### PR DESCRIPTION
**Pull request by doctrino**
# Description

image

Instead of
image

## Checklist

- [ ] Tests added/updated.
- [ ] Run Demo Job Locally.
- [ ] Documentation updated.
- [ ] Changelogs updated in [CHANGELOG.cdf-tk.md](https://github.com/cognitedata/toolkit/blob/main/CHANGELOG.cdf-tk.md).
- [ ] Template changelogs updated in [CHANGELOG.templates.md](https://github.com/cognitedata/toolkit/blob/main/CHANGELOG.templates.md).
- [ ] Version bumped.
  [_version.py](https://github.com/cognitedata/toolkit/blob/main/cognite/cognite_toolkit/_version.py) and
  [pyproject.toml](https://github.com/cognitedata/toolkit/blob/main/pyproject.toml) per [semantic versioning](https://semver.org/).

### CODE REVIEW

File: cognite_toolkit/_cdf_tk/commands/auth.py [hunk 1]
```diff
@@ -119,17 +119,18 @@ def check_auth(
...
-        try:
-            print("Checking projects that the service principal/application has access to...")
-            if len(resp.projects) == 0:
-                raise AuthorizationError(
-                    "The service principal/application configured for this client does not have access to any projects."
-                )
-            projects = ""
-            projects = projects.join(f"  - {p.url_name}\n" for p in resp.projects)
-            print(projects[0:-1])
-        except Exception as e:
```
Comment (doctrino) [inline cognite_toolkit/_cdf_tk/commands/auth.py:131]: This try .. except block is bad as the only exception that can happen is the AuthorizationError which will be caught and replaced by a worse more generic error message.

File: cognite_toolkit/_cdf_tk/commands/auth.py [hunk 2]
```diff
@@ -119,17 +119,18 @@ def check_auth(
...
             )
-        try:
-            print("Checking projects that the service principal/application has access to...")
-            if len(resp.projects) == 0:
-                raise AuthorizationError(
-                    "The service principal/application configured for this client does not have access to any projects."
-                )
-            projects = ""
-            projects = projects.join(f"  - {p.url_name}\n" for p in resp.projects)
-            print(projects[0:-1])
```
Comment (doctrino) [inline cognite_toolkit/_cdf_tk/commands/auth.py:130]: Note that this is is the same as these three lines
```python
print("\n".join(f"  - {p.url_name}" for p in token_inspection.projects))
```

File: cognite_toolkit/_cdf_tk/commands/auth.py [hunk 3]
```diff
@@ -119,17 +119,18 @@ def check_auth(
...
-        except Exception as e:
-            raise AuthorizationError(f"Failed to process project information from inspect()\n{e}")
+
+        print("Checking projects that the service principal/application has access to...")
+        if len(token_inspection.projects) == 0:
+            raise AuthorizationError(
+                "The service principal/application configured for this client does not have access to any projects."
+            )
+        print("\n".join(f"  - {p.url_name}" for p in token_inspection.projects))
```
Comment (doctrino) [inline cognite_toolkit/_cdf_tk/commands/auth.py:128]: This is refactoring

File: cognite_toolkit/_cdf_tk/commands/auth.py [hunk 4]
```diff
@@ -119,17 +119,18 @@ def check_auth(
...
+        if auth_vars.project not in {p.url_name for p in token_inspection.projects}:
+            raise AuthorizationError(
+                f"The service principal/application configured for this client does not have access to the CDF_PROJECT={auth_vars.project!r}."
+            )
```
Comment (doctrino) [inline cognite_toolkit/_cdf_tk/commands/auth.py:132]: This is the new error

File: cognite_toolkit/_cdf_tk/commands/auth.py [hunk 5]
```diff
@@ -108,8 +108,8 @@ def check_auth(
             # Using the token/inspect endpoint to check if the client has access to the project.
             # The response also includes access rights, which can be used to check if the client has the
             # correct access for what you want to do.
-            resp = ToolGlobals.client.iam.token.inspect()
-            if resp is None or len(resp.capabilities) == 0:
+            token_inspection = ToolGlobals.client.iam.token.inspect()
```
Comment (doctrino) [inline cognite_toolkit/_cdf_tk/commands/auth.py:111]: Renaming to make it more understandable

**Review by ronpal — APPROVED** [review]
🫧

**Pull request merged**